### ISSUE TITLE
SampleFill faulty rendering on different browsers

### ISSUE THREAD

**Issue by maeneak** [astrosat/windgl#18]
Tiles are not rendering at certain positions and zoom levels on different browsers. Seems to be fairly random. Can be reproduced using the current demo at https://astrosat.github.io/windgl/

**Comment by maeneak** [comment]
Im fairly sure this is relating to the tiling. When the tiles do display they display correctly along all longitudes but only on 90deg section along the latitude. Strangely the latitudes are different on different browsers. 90N to 0 in firefox and 45N to 45S in chrome and edge. My guess would be computeLoadableTiles() as it is overriden in particles which is functioning as expected.

**Comment by maeneak** [comment]
There's quite a lot going wrong with this. I've identified definite issues with the tile calculations in layer.computeVisibleTiles() but even when fixing this the problem persists. I'm convinced this is a shader issue, possibly relating to the reprojection functions, but my GLSL skills are sketchy at best.

**Comment by gampleman** [comment]
Yeah I've noticed that as well. TBH I'm fairly convinced that some of these issues might even be in mapbox itself, although it's hard to prove. I've even seen inconsistent behavior based on different screen sizes. 

As a workaround, it's probably best to specify a reasonably high min-zoom.

**Comment by maeneak** [comment]
Your work is amazing dude and its helped me immensely. I've been studying [a similar repo](https://github.com/xiaoiver/custom-mapbox-layer) that uses actual mapbox-gl classes for tiling. I'm working on merging the concepts of both repos for my project and ill be sure to share the repo if i can get something up and running. I've also managed to setup a simple MapServer WMS installation that will serve 256x256 tiles/textures on-the-fly from a single encoded PNG at all zoom levels using bi-linear interpolation. Ill post once i get it online.

**Comment by maeneak** [comment]
I can confirm this bug is in the Projections in the Shaders. Using the tilejson above and changing the vertex position to worldCoordsWGS84 from worldCoordsMerc has fixed it. Might be worth testing with the current version.
```
    v_tex_pos = worldCoordsWGS84;
    gl_Position = u_matrix * vec4(worldCoordsWGS84, 0, 1);
```

**Comment by maeneak** [comment]
Testing with the original tile scheme after removing the mercator projection does fix the missing tile issue but the projection is out. Im guessing either the GLSL projection functions or the u_matrix are the culprit

**Comment by maeneak** [comment]
So i managed to find a pretty nifty little hole/hack in the mapbox-gl source code which allows me to use the built in mapbox tile system to feed textures to custom shaders from a custom layer. I've created a [repo with a simple demo](https://github.com/maeneak/mapbox-texture-layer), this would be a good fit with this project and would eliminate many of the alignment and browser related issues. Mapbox takes care of all texture loading, tile matrix calculations, projection and tile caching which will reduce the code base significantly.
Ill be using this method in my own project but will try to throw some time at a branch of the current master and post if i make some progress. This would obviously require a revamp of the py generator code (not my forte) but ill try to use static tiles from a WMS source as test data.

**Comment by gampleman** [comment]
I found those hacks as well, but in this case we are using unprojected data, so the tiles end up being different.

**Comment by maeneak** [comment]
Ahh ok of course. I'm using a tile server which is doing it on the fly. I'm sure there would be plenty of py modules that could do the projection in pre-processing? Anyway ill leave it for now 👍

**Comment by gampleman** [comment]
Of course, but some of the math seemed easier to do correctly in WGS84 than in Mercator. Having now done the project, I think I might reconsider that decision, but at the time it seemed like a reasonable choice.
